CMakeLists.txt

```diff
@@ -160,7 +207,6 @@ if(ASIO_ENABLE_INSTALL)
     BINARY_DIR ${PROJECT_BINARY_DIR}
     INCLUDE_DIR ${asio_SOURCE_DIR}/asio/include
     INCLUDE_DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    INCLUDE_HEADER_PATTERN "*.hpp"
```

Comment:
Shouldn't it stay when not interface?

Reply:
Yes, however generator expression does not work here.
Therefore, wait for https://github.com/TheLartians/PackageProject.cmake/issues/44 to be fixed :/

Reply:
This is an issue if we are loosing something here.
Maybe it is possible to do something based on the TYPE of the library? Or even disable installation with a warning if target is INTERFACE?

Reply:
By default `PackageProject` includes all files (`*`), see https://github.com/TheLartians/PackageProject.cmake/blob/5f51898d02176e2dcf4b07d773062bdd27e00180/CMakeLists.txt#L175
There are only two useless files included: `include/Makefile.am` and `include/.gitignore` which is not a big deal 👌

Reply:
Ok then this is acceptable